static/src/javascripts/projects/common/modules/navigation/newSticky.js

```diff
@@ -0,0 +1,92 @@
+                    mediator.on('modules:commercial:dfp:rendered', function (event) {
+                        if (event.slot.getSlotElementId() === adId) {
+                            resolve();
+                        }
+                    });
+                });
+
+                var getLatestAdHeight = function () {
+                    return fastdom.read(function () {
+                        var adSlotPadding = parseInt($('#' + adId, $adBanner).css('padding-bottom')) * 2;
+                        // We must read the iframe attribute height to avoid reading the clientHeight mid-transition
+                        var iframeHeightStr = $('iframe', $adBanner).attr('height');
+                        var newAdHeight = Number(iframeHeightStr) + adSlotPadding;
+                        return iframeHeightStr ? newAdHeight : oldAdHeight;
+                    });
+                };
+
+                var render = function () {
+                    getLatestAdHeight().then(function (adHeight) {
+                        fastdom.read(function () {
+                            var scrollY = window.scrollY;
+                            fastdom.write(function () {
+                                // Reset
+                                $header.css('margin-top', '');
+                                $adBanner.css({
+                                    'position': '',
+                                    'top': ''
+                                });
+
+                                // Set
+                                if (scrollY > 0) {
+                                    if (scrollY > headerHeight) {
+                                        $adBanner.css({
+                                            'position': 'absolute',
+                                            'top': headerHeight + 'px'
+                                        });
+                                    } else {
+                                        $adBanner.css('position', 'fixed');
+                                    }
+                                    $header.css('margin-top', adHeight + 'px');
+                                }
+                            });
+                        });
+                    });
+                };
+
+                //
+                // Side effects
+                //
+
+                mediator.on('window:throttledScroll', render);
+                render();
+                topAdRenderedPromise.then(render);
+
+                topAdRenderedPromise
+                    .then(getLatestAdHeight)
+                    .then(function (newAdHeight) {
+                        fastdom.read(function () {
+                            var diff = newAdHeight - oldAdHeight;
+
+                            if (window.scrollY !== 0) {
+                                window.scrollTo(window.scrollX, window.scrollY + diff);
+                            }
+                        });
+                    });
```

Comment:
Because if the ad wasn't sticky then there would be no need to adjust the margin on the header and therefore no need to therefore adjust the scroll position.